src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -0,0 +1,257 @@
+                    ArrayBuilder<TextDocumentEdit> textDocumentEdits,
+                    ApplyChangesOperation applyChangesOperation,
+                    Solution solution,
+                    IEnumerable<DocumentId> changedDocuments,
+                    Func<DocumentId, T?> getNewDocumentFunc,
+                    Func<DocumentId, T?> getOldDocumentFunc,
+                    CancellationToken cancellationToken)
+                    where T : TextDocument
+                {
+                    foreach (var docId in changedDocuments)
+                    {
+                        var newDoc = getNewDocumentFunc(docId);
+                        var oldDoc = getOldDocumentFunc(docId);
+                        if (oldDoc == null || newDoc == null)
+                        {
+                            continue;
+                        }
+
+                        var oldText = await oldDoc.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                        var newText = await newDoc.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+                        var textChanges = newText.GetTextChanges(oldText).ToList();
+
+                        var edits = textChanges.Select(tc => ProtocolConversions.TextChangeToTextEdit(tc, oldText)).ToArray();
+                        var documentIdentifier = new VersionedTextDocumentIdentifier() { Uri = newDoc.GetURI() };
+                        textDocumentEdits.Add(new TextDocumentEdit() { TextDocument = documentIdentifier, Edits = edits.ToArray() });
+                    }
+                }
+            }
+
+            static LSP.Command SetCommand(VSCodeAction codeAction, CodeActionResolveData data) => new LSP.Command
+            {
+                CommandIdentifier = CodeActionsHandler.RunCodeActionCommandName,
+                Title = codeAction.Title,
+                Arguments = new object[] { data }
+            };
+        }
+
+        internal static CodeAction? GetCodeActionToResolve(string distinctTitle, IEnumerable<CodeAction> codeActions)
+        {
+            // Searching for the matching code action. We compare against the distinct title (e.g. "Suppress IDExxxxNone")
+            // instead of the regular title (e.g. "None") since there's a chance that multiple code actions may have
+            // the same regular title.
+            CodeAction? codeActionToResolve = null;
+            foreach (var c in codeActions)
+            {
+                var action = CheckForMatchingAction(c, distinctTitle, currentTitle: "");
+                if (action != null)
+                {
+                    codeActionToResolve = action;
+                    break;
+                }
+            }
+
+            return codeActionToResolve;
+        }
+
+        private static CodeAction? CheckForMatchingAction(CodeAction codeAction, string goalTitle, string currentTitle)
+        {
+            if (currentTitle + codeAction.Title == goalTitle)
```

Comment:
introduce local for `currentTitle + codeAction.Title`.  you compute it several times (including in a loop below).

Reply:
in general, when appending stuff like this it's nice to have some sort of delimeter between things so that you don't end up with words all jammed together.  Insetad, you get a nice broken up sequence of pieces that make the hierarchy clear.